client/src/main/java/ru/edu/spbstu/client/controllers/ChatFormController.java

```diff
@@ -0,0 +1,209 @@
+            @Override
+            public void changed(ObservableValue<? extends Chat> observableValue, Chat item, Chat t1) {
+                var curr=chatsListView.getSelectionModel().getSelectedItem();
+                //System.out.println(curr.toString());
+
+                try {
+                    messageList=service.getMessages(curr.getId(),1);
+                } catch (IOException e) {
+                    messageList=new ArrayList<>();
+                }
+                messagesListView.setItems(FXCollections.observableList(messageList));
+
+                //TODO update messageListView
+            }
+        });
+        //chatsListView.init();
+
+        //chatsListView.setItems(FXCollections.observableArrayList(chatList));
+
+    }
+
+
+    private void logOutAction()
+    {
+        currStage.close();
+        primaryStage.show();
+    }
+    public void logOutMouseClick(ActionEvent actionEvent) {
+        logOutAction();
+    }
+
+    public void setPrimaryStage(Stage primaryStage) {
+        this.primaryStage = primaryStage;
+    }
+
+    public void setCurrStage(Stage currStage) {
+        this.currStage = currStage;
+    }
+    void update() {
+        try {
+            chatList = service.getChats(1);
+        }
+        catch(IOException e)
+        {
+            showError("Ошибка при поулчении чатов" +e.getMessage()+" !");
+            return;
+        }
+
+        chatsListView.setItems(FXCollections.observableArrayList(chatList));
+    }
+
+    public void addChatButtonClick(ActionEvent actionEvent) throws IOException {
+
+       // service.addChat(newChatTextBox.getText());
+        FXMLLoader fmxlLoader = new FXMLLoader(getClass().getResource("/fxmls/create_chat_form.fxml"));
+        Parent window = (Pane) fmxlLoader.load();
+        var conC = fmxlLoader.<CreateChatFormController>getController();
+        Scene scene = new Scene(window,800,800);
+
+        conC.setCredentials(this.service.getCredentialsProvider(),this.service.getLogin());
+
+
+        Stage nstage= new Stage();
+        nstage.setScene(scene);
+        nstage.setTitle("Create_chat");
+        conC.setCurrStage(nstage);
+        conC.setPrevController(this);
+        conC.setPrimaryStage(this.currStage);
+        conC.init();
+
+        nstage.show();
+        chatList=service.getChats(1);
+        //update();
+    }
+
+
+
+
+    public void findChatsEvent(KeyEvent keyEvent) {
+        String name=newChatTextBox.getText();
+
+        List<Chat> temp =service.find(name);
+    }
+
+
+
+    public void ProfileMouseClick(ActionEvent actionEvent) throws IOException {
+        FXMLLoader fmxlLoader = new FXMLLoader(getClass().getResource("/fxmls/profile_form.fxml"));
+        Parent window = (Pane) fmxlLoader.load();
+        var conC = fmxlLoader.<ChatFormController>getController();//TODO поменять
+        Scene scene = new Scene(window,700,700);
+
+        conC.setCredentials(this.service.getCredentialsProvider(),this.service.getLogin());
```

Comment:
И здесь можно без this
```suggestion
        conC.setCredentials(service.getCredentialsProvider() ,service.getLogin());
```